backend_new/prisma/schema.prisma

```diff
@@ -0,0 +1,1025 @@
+  totalAvailable                 Int?                            @map("total_available")
+  priorityTypeId                 String?                         @map("priority_type_id") @db.Uuid
+  id                             String                          @id() @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
+  listingId                      String?                         @map("listing_id") @db.Uuid
+  bathroomMin                    Decimal?                        @map("bathroom_min") @db.Decimal
+  bathroomMax                    Decimal?                        @map("bathroom_max") @db.Decimal
+  openWaitlist                   Boolean                         @default(true) @map("open_waitlist")
+  sqFeetMin                      Decimal?                        @map("sq_feet_min") @db.Decimal
+  sqFeetMax                      Decimal?                        @map("sq_feet_max") @db.Decimal
+  listings                       listings?                       @relation(fields: [listingId], references: [id], onDelete: NoAction, onUpdate: NoAction)
+  unitAccessibilityPriorityTypes UnitAccessibilityPriorityTypes? @relation(fields: [priorityTypeId], references: [id], onDelete: NoAction, onUpdate: NoAction)
+  unitGroupAmiLevels             UnitGroupAmiLevels[]
+  unitTypes                      UnitTypes[]
+
+  @@map("unit_group")
+}
+
+model UnitGroupAmiLevels {
+  id                           String                           @id() @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
+  amiPercentage                Int?                             @map("ami_percentage")
+  monthlyRentDeterminationType MonthlyRentDeterminationTypeEnum @map("monthly_rent_determination_type")
+  percentageOfIncomeValue      Decimal?                         @map("percentage_of_income_value") @db.Decimal
+  amiChartId                   String?                          @map("ami_chart_id") @db.Uuid
+  unitGroupId                  String?                          @map("unit_group_id") @db.Uuid
+  flatRentValue                Decimal?                         @map("flat_rent_value") @db.Decimal
+  unitGroup                    UnitGroup?                       @relation(fields: [unitGroupId], references: [id], onDelete: NoAction, onUpdate: NoAction)
+  amiChart                     AmiChart?                        @relation(fields: [amiChartId], references: [id], onDelete: NoAction, onUpdate: NoAction)
+
+  @@map("unit_group_ami_levels")
+}
+
+model UserPreferences {
+  sendEmailNotifications Boolean      @default(false) @map("send_email_notifications")
+  sendSmsNotifications   Boolean      @default(false) @map("send_sms_notifications")
+  userId                 String       @id() @unique() @map("user_id") @db.Uuid
+  favoriteIds            String[]     @default([]) @map("favorite_ids")
+  userAccounts           UserAccounts @relation(fields: [userId], references: [id], onDelete: Cascade, onUpdate: NoAction)
+  favoritesListings      listings[]
```

Comment:
Should this be a relation?

Reply:
yeah so this is a many to many relationship (many listings can be tied to many users and the reverse)

so this is how that relation is defined in prisma